subsys/fs/fs.c

```diff
@@ -720,6 +720,54 @@ int fs_mount(struct fs_mount_t *mp)
 	return rc;
 }
 
+#if defined(CONFIG_FILE_SYSTEM_MKFS)
+
+int fs_mkfs(struct fs_mount_t *mp)
+{
+	const struct fs_file_system_t *fs;
+	int rc = -EINVAL;
+
+	/* Do all the mp checks prior to locking the mutex on the file
+	 * subsystem.
+	 */
+	if ((mp == NULL) || (mp->mnt_point == NULL)) {
+		LOG_ERR("mount point not initialized!!");
+		return -EINVAL;
+	}
+
+	if (sys_dnode_is_linked(&mp->node)) {
+		LOG_ERR("file system already mounted!!");
+		return -EBUSY;
+	}
+
+	k_mutex_lock(&mutex, K_FOREVER);
+
+	/* Get file system information */
+	fs = fs_type_get(mp->type);
+	if (fs == NULL) {
+		LOG_ERR("requested file system type not registered!!");
+		rc = -ENOENT;
+		goto mount_err;
+	}
+
+	CHECKIF(fs->mkfs == NULL) {
+		LOG_ERR("fs type %d does not support mkfs", mp->type);
+		rc = -ENOTSUP;
+		goto mount_err;
+	}
+
+	rc = fs->mkfs(mp);
```

Comment:
mkfs should respect `FS_MOUNT_FLAG_NO_FORMAT`, if such flag is set to mount point  mkfs should return `-EROFS`.

Reply:
Now I also check for that flag.